### PR TITLE
Fix interlace timing and modeling.

### PR DESCRIPTION
**Pull request by scarybeasts**
- Odd and even interlace frames are now the same length.
- The even frame dummy raster is a vertical adjust scanline.

### CODE REVIEW

File: video.js [hunk 1]
```diff
@@ -362,26 +365,20 @@ define(['./teletext', './utils'], function (Teletext, utils) {
...
                 }
 
-                this.bitmapY++;
+                // The CRT vertical beam speed is constant, so this is actually
+                // an approximation that works if hsyncs are spaced evenly.
+                this.bitmapY += 2;
+
                 // If no VSync occurs this frame, go back to the top and force a repaint
-                if (this.bitmapY >= 384) {
+                if (this.bitmapY >= 768) {
```
Comment (mattgodbolt) [inline video.js:373]: I've been suspicious of this constant a good long while: what was the rationale for bumping it? (did you find a way to expose this?)

Thanks again for your tireless work on this!

File: video.js [hunk 2]
```diff
@@ -362,26 +365,20 @@ define(['./teletext', './utils'], function (Teletext, utils) {
                     this.bitmapX -= 4;
                 }
 
-                this.bitmapY++;
+                // The CRT vertical beam speed is constant, so this is actually
+                // an approximation that works if hsyncs are spaced evenly.
+                this.bitmapY += 2;
```
Comment (mattgodbolt) [inline video.js:370]: Makes perfect sense!

File: video.js (hunk 1 above)
Comment (scarybeasts) [reply video.js:373]: This constant in some sense hasn't "changed". I removed "this.renderY" in order to make the code simpler and less fragile to change. Now there's only the one variable, "this.bitmapY" for tracking where the CRT vertical beam position is within the canvas.
As part of that change, "this.bitmapY" doubled in value, including this constant.
The constant now does represent PAL lines, so IMHO it's a bit more easy to see that it's a little more than ~624, the standard number of PAL lines per PAL frame.

Unfortunately I have no idea if there's a better value for the constant!

Comment (mattgodbolt) [reply video.js:373]: Thanks fella, got it :)

**Pull request merged**